spec/text-editor-component-spec.js

```diff
@@ -896,6 +896,39 @@ describe('TextEditorComponent', () => {
       expect(component.getLineNumberGutterWidth()).toBe(originalLineNumberGutterWidth)
     })
 
+    it('gracefully handles edits that change the maxScrollTop by causing the horizontal scrollbar to disappear', async () => {
+      const rowsPerTile = 1
+      const {component, element, editor} = buildComponent({rowsPerTile, autoHeight: false})
+
+      await setEditorHeightInLines(component, 1)
+      await setEditorWidthInCharacters(component, 7)
+
+      element.focus()
+      component.setScrollTop(component.measurements.lineHeight)
+
+      component.scheduleUpdate()
+      await component.getNextUpdatePromise()
+
+      editor.setSelectedBufferRange(new Range(Point(0,1),Point(12,2)))
```

Comment:
It's fine to express a range as nested arrays like `[[0, 1], [12, 2]]`, especially in tests.